Client/Users/index.ts

```diff
@@ -19,25 +19,58 @@ export class Users extends Collection<model.UserResponse, model.UserSearchReques
 		const result = await this.connection.remove<model.UserResponse>(`users/${username}`)
 		return result
 	}
-	async getAllUsers(): Promise<model.UserResponse[] | model.ErrorResponse> {
-		const result = await this.connection.get<model.UserResponse[]>(`users?size=500&sort=username`)
+	async getAllUsers(): Promise<model.UserResponse[] | model.ErrorResponse>
+	async getAllUsers(
+		withCount: boolean
+	): Promise<{ list: model.UserResponse[]; totalCount: number } | model.UserResponse[] | model.ErrorResponse>
+	async getAllUsers(
+		withCount?: boolean
+	): Promise<{ list: model.UserResponse[]; totalCount: number } | model.UserResponse[] | model.ErrorResponse> {
+		let result
+		result = await this.connection.get<{ list: model.UserResponse[]; totalCount: number }>(
+			`users?size=500&sort=username`
+		)
+		if (!model.ErrorResponse.is(result) && !withCount)
+			result = result.list
 		return result
 	}
 	async getCategory(): Promise<string[] | model.ErrorResponse> {
 		const result = await this.connection.get<string[]>(`users/category`)
 		return result
 	}
-	async getRolesets(): Promise<model.RolesetResponse[] | model.ErrorResponse> {
-		const result = await this.connection.get<model.RolesetResponse[]>(`rolesets`)
+	async getRolesets(): Promise<model.RolesetResponse[] | model.ErrorResponse>
+	async getRolesets(
+		withCount: boolean
+	): Promise<{ list: model.RolesetResponse[]; totalCount: number } | model.RolesetResponse[] | model.ErrorResponse>
+	async getRolesets(
+		withCount?: boolean
+	): Promise<{ list: model.RolesetResponse[]; totalCount: number } | model.RolesetResponse[] | model.ErrorResponse> {
+		let result
+		result = await this.connection.get<{ list: model.RolesetResponse[]; totalCount: number }>(`rolesets`)
+		if (!model.ErrorResponse.is(result) && !withCount)
+			result = result.list
 		return result
 	}
 	async getUser(username: string): Promise<model.UserResponse | model.ErrorResponse> {
 		const result = await this.connection.get<model.UserResponse>(`users/${username}`)
 		return result
 	}
-	async getUsersActiveRoles(username: string, token: string): Promise<string[] | model.ErrorResponse> {
+	async getUsersActiveRoles(username: string, token: string): Promise<string[] | model.ErrorResponse>
+	async getUsersActiveRoles(
+		username: string,
+		token: string,
+		withCount: boolean
+	): Promise<{ list: string[]; totalCount: number } | string[] | model.ErrorResponse>
+	async getUsersActiveRoles(
+		username: string,
+		token: string,
+		withCount?: boolean
+	): Promise<{ list: string[]; totalCount: number } | string[] | model.ErrorResponse> {
```

Comment:
any reason for this to take the token?

Reply:
for the way this is written in ui, yes. Not sure if we should change into the other way where we set the token through the components

Reply:
just not sure why it needs the token when nothing else does?